src/test/kotlin/Generators.kt

```diff
@@ -0,0 +1,35 @@
+import io.kotlintest.properties.Gen
+
+class CellGen : Gen<Cell> {
+    override fun constants(): Iterable<Cell> = emptyList()
+
+    override fun random(): Sequence<Cell> =
+        generateSequence {
+            val random = Gen.bool().random().first()
+            if (random) Cell.Alive() else Cell.Dead()
+        }
+}
+
+class UniverseGen(private val center: Cell, private val aliveNeighbors: Int) : Gen<Universe> {
+
+    override fun constants(): Iterable<Universe> = emptyList()
+
+    override fun random(): Sequence<Universe> = generateSequence {
+        val alive = (1..aliveNeighbors).map { Cell.Alive() }.toList()
+        val dead = (1..(8 - aliveNeighbors)).map { Cell.Dead() }.toList()
+        val all = alive + dead
+
+        all.shuffled().toGrid(center)
+    }
+
+    private fun List<Cell>.toGrid(center: Cell): List<List<Cell>> =
+        listOf(
+            listOf(get(0), get(1), get(2)),
+            listOf(get(3), center, get(4)),
+            listOf(get(5), get(6), get(7))
+        )
+}
+
+fun universeGenWith2or3AliveNeighbours(cell: Cell) = UniverseGen(cell, Gen.choose(2, 3).random().first())
```

Comment:
I think this will generate only universes of the first size you create. To make it more generic and get a world of a different size each time, you need to do something like (I'm guessing syntax):

```kotlin
Gen.choose(2, 3).flatMap { UniverseGen(cell, it }
```

The rest of the generators should be the same.

Reply:
Good point, I thought the creation function got called once per test execution. Fixed all the utility functions to match the style you suggest just in case.